### ISSUE TITLE
Commits get made, even if the gem being updated had it's version stay the same

### ISSUE THREAD

**Issue by sbleon** [singlebrook/bunup#11]
```
(1/40) Updating actioncable 5.2.0 -> 5.2.1
ERROR: Bundler tried to update actioncable but is version stayed the same
```

I end up with a commit message that says "actioncable 5.2.1 (was 5.2.0)", but it actually only updates dependencies of actioncable.

**Comment by shanecav84** [comment]
We might be able to do two bunup runs: one with `--only-explicit` for Bundler 1.17+ to update gems specifically listed in the Gemfile; and a run without the `--only-explicit` to get everything.

**Comment by sbleon** [comment]
Yeah, I like that idea! Hit the top-level dependencies first, then the transitive dependencies that are still behind.

**Comment by shanecav84** [comment]
We could bring in another dependency, [bundler-stats](https://github.com/jmmastey/bundler-stats), which has some nice tools to get a Gemfile's dependency information and can output parseable JSON.

**Comment by sbleon** [comment]
Why not just use the `--only-explicit` flag you mentioned? I'm OK if that
behavior only works with Bundler 1.17+.

**Comment by shanecav84** [comment]
Just that it won't support less than Bundler 1.17 and helping people upgrade from really old versions of stuff is a use case for this gem. I'm find doing the double-run for just Bundler 1.17 though.
